client/cmd/resource/plan.go

```diff
@@ -0,0 +1,233 @@
+			}
+			p.logger.Info(msg)
+		}
+	}
+	return p.saveFile(mergedPlans)
+}
+
+func (p *planCommand) describePlanFromDirectory(ctx context.Context, directory string) (*plan.Plan, error) {
+	var (
+		namespaceName, datastoreType string
+		err                          error
+		sourceRaw, destinationRaw    []byte
+		fileName                     = filepath.Join(directory, resourceFileName)
+	)
+
+	for _, namespace := range p.clientConfig.Namespaces {
+		for _, datastore := range namespace.Datastore {
+			if strings.HasPrefix(directory, datastore.Path) {
+				namespaceName = namespace.Name
+				datastoreType = datastore.Type
+				break
+			}
+		}
+	}
+	if namespaceName == "" {
+		return nil, fmt.Errorf("failed to find namespace specified on directory: %s", directory)
+	}
+
+	destinationRaw, err = p.repository.GetFileContent(ctx, p.projectID, p.destinationRef, fileName)
+	if err != nil {
+		return nil, errors.Join(err, fmt.Errorf("failed to get file with ref: %s and directory %s", p.destinationRef, directory))
+	}
+	sourceRaw, err = p.repository.GetFileContent(ctx, p.projectID, p.sourceRef, fileName)
+	if err != nil {
+		return nil, errors.Join(err, fmt.Errorf("failed to get file with ref: %s and directory %s", p.sourceRef, directory))
+	}
+
+	var sourceSpec, destinationSpec model.ResourceSpec
+	if err = yaml.Unmarshal(sourceRaw, &sourceSpec); err != nil {
+		return nil, errors.Join(err, errors.New("failed to unmarshal source resource specification"))
+	}
+	if err = yaml.Unmarshal(destinationRaw, &destinationSpec); err != nil {
+		return nil, errors.Join(err, errors.New("failed to unmarshal destination resource specification"))
+	}
+
+	resourcePlan := &plan.Plan{ProjectName: p.clientConfig.Project.Name, NamespaceName: namespaceName, Kind: plan.KindResource}
+	if p.isOperationCreate(sourceSpec, destinationSpec) {
+		resourcePlan.KindName = fmt.Sprintf("%s:%s", datastoreType, destinationSpec.Name)
+		resourcePlan.Operation = plan.OperationCreate
+	} else if p.isOperationDelete(sourceSpec, destinationSpec) {
+		resourcePlan.KindName = fmt.Sprintf("%s:%s", datastoreType, sourceSpec.Name)
+		resourcePlan.Operation = plan.OperationDelete
+	} else {
+		resourcePlan.KindName = fmt.Sprintf("%s:%s", datastoreType, destinationSpec.Name)
+		resourcePlan.Operation = plan.OperationUpdate
+	}
+
+	return resourcePlan, nil
+}
+
+// isOperationCreate return true when destinationSpec is exists, but sourceSpec is missing
+func (*planCommand) isOperationCreate(sourceSpec, destinationSpec model.ResourceSpec) bool {
+	return len(sourceSpec.Name) == 0 && len(destinationSpec.Name) > 0
+}
+
+// isOperationCreate return true when sourceSpec is exists, but destinationSpec is missing
+func (*planCommand) isOperationDelete(sourceSpec, destinationSpec model.ResourceSpec) bool {
+	return len(sourceSpec.Name) > 0 && len(destinationSpec.Name) == 0
+}
+
+func (*planCommand) appendDirectory(directory string, directoryExists map[string]bool, fileDirectories []string) []string {
```

Comment:
what is the purpose of this appendDirectory can we have this function with a business meaning